### DISCUSSION TITLE
Nuxt3/Vue3 in production is reporting many instances of Cannot find static asset /_nuxt/<filename>.<hash>.js

### DISCUSSION THREAD

**Discussion by technoY2K** [nuxt/nuxt#25703] [Questions]
Hello everyone. We have a website deployed and are getting thousands of reports that assets cannot be found. The assets are JavaScript chunks, CSS, etc. The site works with no issues, however, the server instance is throwing errors which we pinned down to Nitro server code: My hypothesis is that for some reason the server is looking for older hashed files which wouldn't exist since every time we deploy, new hashed files are uploaded. 

```
if (!asset) {
    if (isPublicAssetURL(id)) {
      removeResponseHeader(event, "Cache-Control");
      throw createError$1({
        statusMessage: "Cannot find static asset " + id,
        statusCode: 404
      });
    }
    return;
```

My question is 2 fold.

1. Could someone possibly explain what is happening? 

2. What are some steps we could take to possibly mitigate these errors.

**Comment by TheAlexLichter** [comment]
> My hypothesis is that for some reason the server is looking for older hashed files which wouldn't exist since every time we deploy, new hashed files are uploaded.

That sounds reasonable, yes.

> Could someone possibly explain what is happening?

Most likely that, after a new deploy, current visitors still request older static assets of your site 

> What are some steps we could take to possibly mitigate these errors.

Save them e.g. via CDN and cache them on the server forever.

**Reply by technoY2K** [reply]
Hello, do you mean to save the old hashed files along side the new deployed files that are created under `.output`?

**Reply by zestlee1106** [reply]
Hello, I have the same issue with my project, did you solve it?

**Reply by existe-deja** [reply]
I'm working on this solution and you should take a look at https://nuxt.com/docs/api/composables/use-runtime-config#appcdnurl

**Reply by technoY2K** [reply]
Hey @zestlee1106 , we ended up just switching over to AWS Amplify to avoid the issue all together.

**Comment by narita-takeru** [comment]
Hello.
I also encountered the same problem.

The problem seems to be resolved by adding hash to the query string instead of the file name in the rollupOptions of config.
Keep the file name fixed.

But why does nuxt add hash to file names by default?

**Reply by TheAlexLichter** [reply]
To cache-bust the files when their content changes.
